src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -594,15 +595,16 @@ boolean isEmpty() {
      * @param language the language in which it is written
      * @param context any additional information about how where or by whom this is being configured
      * @param approveIfAdmin indicates whether script should be approved if current user has admin permissions
+     * @param ignoreAdmin indicates whether auto approval should be ignored, regardless of any configurations.
      * @return {@code script}, for convenience
      */
-    public synchronized String configuring(@NonNull String script, @NonNull Language language, @NonNull ApprovalContext context, boolean approveIfAdmin) {
+    public synchronized String configuring(@NonNull String script, @NonNull Language language, @NonNull ApprovalContext context, boolean approveIfAdmin, boolean ignoreAdmin) {
         final ConversionCheckResult result = checkAndConvertApprovedScript(script, language);
         if (!result.approved) {
-            if (!Jenkins.get().isUseSecurity() || 
+            if (!Jenkins.get().isUseSecurity() ||
```

Comment:
The boolean logic here has become too complex and unreadable, so it is very difficult to grok if the added logic is sound.

But why do you need the new parameter at all? From the description it seems like it should suffice to just set `approveIfAdmin` to `false`?

Reply:
Hey @rsandell, thanks for the review!

I agree that it is too complex.
The reason for adding this can be found [here](https://github.com/jenkinsci/script-security-plugin/pull/546#issuecomment-1827751225).

Does this answer your question? If yes, I would improve the readbility of the boolean. If not, let's further discuss its need.

Reply:
Sorry I couldn't find @Wadeck 's comment so I am still a bit lost as to why, though I trust that there is a legitimate reeason for it I just would like to understand that reason :sweat_smile: 

If you could clean up the logic to make it more readable that would be hugely helpful :) Maybe it is just enough to reformat the code?

Reply:
No worries!
https://github.com/jenkinsci/script-security-plugin/pull/300#issuecomment-685495550
There you go :) Last sentence in that comment.

What ignoreAdmin does can be seen [here](https://github.com/jenkinsci/script-security-plugin/commit/3a8f7ee16ae579933ee1f6cfaf2ca356fa4875ff#diff-5ed519b7b22dea269dd4d1feeb9ba76ced885593609c49fc25e9eb429e68702fR659). When the using() function is called from a job, it will not cause scripts to be approved automatically. If the admin did set up the script, it would have already been approved on saving the configuration.

It is supposed to prevent the scenario, where an admin starts a job, not knowing that it will approve a potentially dangerous script in it.

I am reformatting it in a sec

Reply:
>  though I trust that there is a legitimate reeason for it I just would like to understand that reason 😅

Yes, please! I am new to this repository's source code. There is a chance that I am completely wrong with that new boolean and since this is security relevant, I totally support that you question it.

Reply:
I tried to make it much more readable, even though not as "fancy" as before. Does this help? @rsandell